### ISSUE TITLE
add faceted counts for dataset and publishers in literature

### ISSUE THREAD

**Issue by gbif-portal** [gbif/portal-feedback#3415]
**add faceted counts for dataset and publishers in literature**

Is it possible to add facets with top 20 counts for 

Dataset `gbifDatasetKey`
Publisher `publishingOrganizationKey`

(similar to what we have for journal and journal publishers)?

This is important when reporting on which datasets get used the most.

-----

Github user: @dnoesgaard
User: [See in registry](https://www.gbif.org/api/feedback/user/06336e8d03e95bcf8d0571ac9acb921c:2528075eebc7dce16f691e1d04f0d7955049ba570cfbc0db8d4987cd021cbb58bd9c1e93dd0ebb8100c8c99f982949ba37875ae7d89be8dd152dcce5552f84ce)
System: Chrome 90.0.4430 / Mac OS X 10.15.7
Referer: https://www.gbif.org/resource/search?contentType=literature
Window size: width 1440 - height 679
[API log](https://private-logs.gbif.org/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2021-05-11T07:26:17.892Z',to:'2021-05-11T07:32:17.892Z'))&_a=(columns:!(_source),filters:!(),index:'3390a910-fcda-11ea-a9ab-4375f2a9d11c',interval:auto,query:(language:kuery,query:''),sort:!()))
[Site log](https://private-logs.gbif.org/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2021-05-11T07:26:17.892Z',to:'2021-05-11T07:32:17.892Z'))&_a=(columns:!(_source),filters:!(),index:'5c73f360-fce3-11ea-a9ab-4375f2a9d11c',interval:auto,query:(language:kuery,query:''),sort:!()))
System health at time of feedback: OPERATIONAL

**Comment by MortenHofft** [comment]
http://api.gbif.org/v1/literature/search?facet=gbifDatasetKey&limit=0
For reporting the API is probably better to use since you can control facet size?

**Comment by dnoesgaard** [comment]
For our internal reporting, absolutely. This wasn't so much intended for me, but for publishers and other stakeholders.

**Comment by MortenHofft** [comment]
Did anyone ask? I agree it sounds useful for them, but it is always interesting to know where the idea comes from.

**Comment by dnoesgaard** [comment]
This came out of an internal discussion.

**Comment by MortenHofft** [comment]
The reason I'm not just jumping and this but ask questions is that it is surprisingly cumbersome to do with our current codebase. It is doable of course, but it isn't just a 10 min task. More like a day to refactor those parts. Let me know if you believe it is worth the effort. 
Personally I would prefer to keep it in mind for the next rewrite unless external people start asking for it.

**Comment by dnoesgaard** [comment]
Ok, I wasn't aware. Is it because we only have uuids as names for datasets and publishers?

**Comment by MortenHofft** [comment]
yeah, in combination with literature not having a version 1 API when we did this, so it is a custom solution that is different from all the other search endpoints.
